contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L1ETHGateway.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: ovm
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Interface Imports */
+import { iOVM_L1ETHGateway } from "../../../iOVM/bridge/tokens/iOVM_L1ETHGateway.sol";
+import { iOVM_L2DepositedERC20 } from "../../../iOVM/bridge/tokens/iOVM_L2DepositedERC20.sol";
+import { iOVM_ERC20 } from "../../../iOVM/precompiles/iOVM_ERC20.sol";
+
+/* Library Imports */
+import { OVM_CrossDomainEnabled } from "../../../libraries/bridge/OVM_CrossDomainEnabled.sol";
+import { Lib_AddressResolver } from "../../../libraries/resolver/Lib_AddressResolver.sol";
+
+/**
+ * @title OVM_L1ETHGateway
+ * @dev The L1 ETH Gateway is a contract which stores deposited ETH that is in use on L2.
+ *
+ * Compiler used: solc
+ * Runtime target: EVM
+ */
+contract OVM_L1ETHGateway is iOVM_L1ETHGateway, OVM_CrossDomainEnabled, Lib_AddressResolver {
+    /********************************
+     * External Contract References *
+     ********************************/
+
+    address public l2ERC20Gateway;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /**
+     * @param _libAddressManager Address manager for this OE deployment
+     */
+    constructor(
+        address _libAddressManager,
+        address _l2ERC20Gateway
+    )
+        OVM_CrossDomainEnabled(address(0)) // overridden in constructor code
+        Lib_AddressResolver(_libAddressManager)
+    {
+        l2ERC20Gateway = _l2ERC20Gateway;
+        messenger = resolve("Proxy__OVM_L1CrossDomainMessenger"); // overrides OVM_CrossDomainEnabled constructor setting because resolve() is not yet accessible
+    }
+
+    /**************
+     * Depositing *
+     **************/
+
+    /**
+     * @dev deposit an amount of the ERC20 to the caller's balance on L2
+     */
+    function deposit() external override payable {
+        _initiateDeposit(msg.sender, msg.sender);
+    }
+
+    /**
+     * @dev deposit an amount of ERC20 to a recipients's balance on L2
+     * @param _to L2 address to credit the withdrawal to
+     */
+    function depositTo(address _to) external override payable {
+        _initiateDeposit(msg.sender, _to);
+    }
```

Comment:
Is it a standard practice to override here just to add payable? / this doesn't seem to have the standard formatting